### PR TITLE
Fixed ray-tracing variant of the checkerboard shader graph node

### PR DESCRIPTION
**Pull request by eturquin**
Implemented a more readable and robust variant for ray tracing.
The previous, slightly ambiguous code was failing with recent NVidia drivers (was always returning the same color).

### CODE REVIEW

**Comment by anisunity** [comment]
Good catch, but i wonder why it was working because the test were not failing

**Review by remi-chapelain — APPROVED** [review]
Looks good to me. Works the same on the user side ✔️

File: com.unity.shadergraph/Editor/Data/Nodes/Procedural/CheckerboardNode.cs [hunk 1]
```diff
@@ -27,12 +27,12 @@ protected override MethodInfo GetFunctionToConvert()
             return
 @"
 {
-    UV = (UV.xy + 0.5) * Frequency;
-    $precision2 distance3 = 4.0 * abs(frac(UV + 0.25) - 0.5) - 1.0;
 #if defined(SHADER_STAGE_RAY_TRACING)
-    int2 alpha = saturate(distance3 * FLT_MAX);
```
Comment (anisunity) [inline com.unity.shadergraph/Editor/Data/Nodes/Procedural/CheckerboardNode.cs:33]: this must be throwing warnings on console

Comment (eturquin) [reply com.unity.shadergraph/Editor/Data/Nodes/Procedural/CheckerboardNode.cs:33]: Well, it's gone now.

Comment (eturquin) [reply com.unity.shadergraph/Editor/Data/Nodes/Procedural/CheckerboardNode.cs:33]: Ah, no it doesn't.

**Review by sebastienlagarde — APPROVED** [review]

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
For SG team: the behavior of the node don't change for regular rasterization, only for raytracing, merging.

**Pull request merged**